### PR TITLE
Extracts the conflict condition into a factory

### LINKED ISSUE #2: Extract booking conflict condition
The booking [conflict condition][1] (used by the [`BookingConflictValidator`] in the booking logic) needs to be extracted into a factory and the validator made to accept the factory as a dependency.

This separates the condition from the actual validator, allowing the conflict condition to change, but keeping the [`BookingConflictValidator`] logic the same.

[1]: https://github.com/RebelCode/rcmod-eddbk-booking-logic/blob/develop/src/BookingConflictValidator.php#L116
[`BookingConflictValidator`]: https://github.com/RebelCode/rcmod-eddbk-booking-logic/blob/develop/src/BookingConflictValidator.php

### PR DESCRIPTION
**Pull request by mecha**
Resolves #2

### CODE REVIEW

File: src/BookingConflictConditionFactory.php [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+                    $b->lit(S::STATUS_APPROVED)
+                ),
+                // Booking status is `scheduled`
+                $b->eq(
+                    $b->ef('booking', 'status'),
+                    $b->lit(S::STATUS_SCHEDULED)
+                )
+            ),
+            // Bookings' service IDs are the same
+            $b->eq(
```
Comment (XedinUnknown) [inline src/BookingConflictConditionFactory.php]: Why do we need to compare service ID? Isn't resource ID enough? I mean that no two bookings can be made for the same resource at the same time, regardless of their service. Or not?

File: src/BookingConflictConditionFactory.php [hunk 2]
```diff
@@ -0,0 +1,138 @@
...
+        // Booking must be able to provide additional data in order to tighten the condition
+        if (!($booking instanceof ContainerInterface)) {
+            return $overlap;
+        }
+
+        return $b->and(
+            $overlap,
+            // Booking status is either of the below:
+            $b->or(
+            // Booking status is `approved`
```
Comment (XedinUnknown) [inline src/BookingConflictConditionFactory.php]: Looks beautiful <3 Except for the indentation of this line ;P

**Review by XedinUnknown — CHANGES_REQUESTED** [review]

File: src/BookingConflictConditionFactory.php (hunk 1 above)
Comment (mecha) [reply src/BookingConflictConditionFactory.php]: Good point. I will amend.

File: src/BookingConflictConditionFactory.php (hunk 2 above)
Comment (mecha) [reply src/BookingConflictConditionFactory.php]: I will fix.

**Pull request merged**